src/provider/store/Actions.ts

```diff
@@ -70,19 +77,26 @@ export const Actions: ActionMap = {
     },
     [Action.REMOVE]: (state: Immutable<RootState>, action: RemoveNotifications): RootState => {
         const {notifications} = action;
+        const database = Injector.get<'DATABASE'>(Inject.DATABASE);
         const idsToRemove = notifications.map(n => {
-            notificationStorage.removeItem(n.id);
             return n.id;
         });
 
+        database.get(CollectionMap.NOTIFICATIONS).delete(idsToRemove);
+
         return {
             ...state,
             notifications: mutable(state.notifications.filter(n => idsToRemove.indexOf(n.id) === -1))
         };
     },
     [Action.TOGGLE_VISIBILITY]: (state: Immutable<RootState>, action: ToggleVisibility): RootState => {
+        const storage = Injector.get<'DATABASE'>(Inject.DATABASE);
         const windowVisible = (action.visible !== undefined) ? action.visible : !state.windowVisible;
-        settingsStorage.setItem('windowVisible', windowVisible);
+
+        storage.get(CollectionMap.SETTINGS).upsert({id: SettingsMap.WINDOW_VISIBLE, value: windowVisible})
+            .catch(error => {
```

Comment:
How is this different from just not having the catch and letting the promise reject normally?

Reply:
I suppose you are right. We dont handle the exception elsewhere so I will remove it from here.

Reply:
Done